sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AbstractSqlParser.scala

```diff
@@ -102,6 +103,10 @@ abstract class AbstractSqlParser extends AbstractParser with ParserInterface {
     }
   }
 
+  override def parseRoutineParam(sqlText: String): StructType = parse(sqlText) { parser =>
+    astBuilder.visitSingleRoutineParamList(parser.singleRoutineParamList())
```

Comment:
shall we follow other methods in this file and wrap it with `withErrorHandling`?